### ISSUE TITLE
Define the data repositories

### ISSUE THREAD

**Issue by chininchu** [chininchu/Dining-Review-API#11]
At this point, you should have defined your various entities that comprise our restaurant dining review app scenario. For this task, let’s provide you with more details about the different scenarios. This will help you identify the kind of queries you may need to make.

User entity-related scenarios:

- [x] As an unregistered user, I want to create my user profile using a display name that’s unique only to me.
- [x] As a registered user, I want to update my user profile. I cannot modify my unique display name.
- [x] As an application experience, I want to fetch the user profile belonging to a given display name.
- [x] As part of the backend process that validates a user-submitted dining review, I want to verify that the user exists, based on the user display name associated with the dining review.

Dining review entity-related scenarios:

- [x] As a registered user, I want to submit a dining review.
- [x] As an admin, I want to get the list of all dining reviews that are pending approval.
- [x] As an admin, I want to approve or reject a given dining review.
- [x] As part of the backend process that updates a restaurant’s set of scores, I want to fetch the set of all approved dining reviews belonging to this restaurant.

Restaurant entity-related scenarios:

- [x] As an application experience, I want to submit a new restaurant entry. Should a restaurant with the same name and with the same zip code already exist, I will see a failure.
- [x] As an application experience, I want to fetch the details of a restaurant, given its unique Id.
- [x] As an application experience, I want to fetch restaurants that match a given zip code and that also have at least one user-submitted score for a given allergy. I want to see them sorted in descending order.

**Comment by chininchu** [comment]
Hi @JMComstock ,

At CodeUp I clearly remembered  when we created dependency injections within the controller class we did it manually, but looking at other people's work it seems to me that it is best practice to use the @AutoWired annotation when creating dependency injections. Apparently, the @Autowired annotation marks the constructor to be used for injections. Can you please explain to me what the difference is between the two?. I have use the @Autowired annotation within my DiningUserService class. Thanks.

**Comment by JMComstock** [comment]
Hello @chininchu, 

First I will explain Manual and Automatic dependency injection...

**Manual Dependency Injection:**
In manual dependency injection, you create the dependencies yourself and provide them to the dependent class, usually through a constructor, setter methods, or other means. This involves managing the creation and passing of dependencies explicitly. This approach can be more verbose and requires you to handle the dependency resolution and management yourself.

 **@Autowired Annotation:**

Spring Framework, which is a popular Java framework, provides an @Autowired annotation that allows you to automatically inject dependencies into your classes. This annotation is placed on a constructor, setter method, or field, and Spring's container takes care of resolving and injecting the dependencies for you. This greatly simplifies the process of managing and injecting dependencies.

_In summary, the key differences are:_

**Manual Injection:** You explicitly create and provide dependencies to a class.
**@Autowired Annotation:** You annotate a constructor, setter method, or field with @Autowired, and Spring automatically resolves and injects the dependencies.
Using @Autowired is generally considered a best practice as it promotes loose coupling between classes, reduces manual wiring, and simplifies the codebase. It also makes your code more maintainable and easier to test because you can easily provide mock dependencies during testing. However, it's important to note that to use @Autowired, you need to have Spring or a similar dependency injection framework integrated into your project.

**Comment by JMComstock** [comment]
``

**Comment by JMComstock** [comment]
`curl -X POST "http://localhost:8080/api/users/create" \
-H "Content-Type: application/json" \
-d '{
  "displayName": "john_doe",
  "city": "New York",
  "state": "NY",
  "zipcode": "10001",
  "interestedInPeanutAllergies": true,
  "interestedInEggAllergies": true,
  "interestedInDairyAllergies": true
}'
`

**Comment by JMComstock** [comment]
I have emailed you a response sir.
